src/Filters/filt.jl

```diff
@@ -151,16 +151,22 @@ end
 
 ## PolynomialRatio
 DF2TFilter(coef::PolynomialRatio{:z,T},
-           state::Vector{S}=zeros(T, max(length(coef.a), length(coef.b))-1)) where {T,S} =
+           state::Vector{S}=zeros(T, max(length(coefa(coef)), length(coefb(coef)))-1)) where {T,S} =
     DF2TFilter{PolynomialRatio{:z,T}, Vector{S}}(coef, state)
 
 function filt!(out::AbstractVector, f::DF2TFilter{<:PolynomialRatio,<:Vector}, x::AbstractVector)
     length(x) != length(out) && throw(ArgumentError("out size must match x"))
 
     si = f.state
+    n = length(si) + 1
     b = coefb(f.coef)
+    if length(b) < n
+        append!(b, zeros(n-length(b)))
+    end
     a = coefa(f.coef)
-    n = length(b)
+    if length(a) < n
+        append!(a, zeros(n-length(a)))
+    end
```

Comment:
Note that `coefb` and `coefa` allocate already, so I aimed at the simple bugfix here, even if it may mean an additional allocation.

If we want to reduce the overhead here, we would need to access `coef.b` and `coef.a` directly, but that will likely make the actual filtering loop slower because indexing them is not as straight-forward (but on the plus-side, includes the zero-padding). Alternative, we could store the `b` and `a` in `DF2TFilter`, but that would require the type to be reorganized a bit.